### PR TITLE
feat: add log-level configuration option

### PR DESCRIPTION
**Pull request by gruyaume**
# Description

Here we add a `log-level` configuration option with a reasonable default value (`info`). The allowed options are the same as in the AMF workload: `debug`, `info`, `warn`, `error`, `fatal`, `panic`.

## Rationale

In most deployments, the log level should be set to `info` to ensure the log quantity is reasonable. 

## Logs

```
2024-12-05T14:21:38.034Z [pebble] Service "amf" starting: /bin/amf --amfcfg /free5gc/config/amfcfg.conf
2024-12-05T14:21:38.044Z [amf] 2024-12-05T14:21:38.044Z	INFO	build/amf.go:24	amf	{"component": "AMF", "category": "App"}
2024-12-05T14:21:38.045Z [amf] 2024-12-05T14:21:38.045Z	INFO	service/init.go:239	AMF Log level is set to [info] level	{"component": "AMF", "category": "Init"}
2024-12-05T14:21:38.045Z [amf] 2024-12-05T14:21:38.045Z	INFO	logger/logger.go:103	set log level: info	{"component": "AMF", "category": "CFG"}
2024-12-05T14:21:38.045Z [amf] 2024-12-05T14:21:38.045Z	INFO	factory/factory.go:136	config version [1.0.0]	{"component": "AMF", "category": "CFG"}
2024-12-05T14:21:38.045Z [amf] 2024-12-05T14:21:38.045Z	INFO	service/init.go:147	Reading Amf related configuration from ROC	{"component": "AMF", "category": "Init"}
2024-12-05T14:21:38.045Z [amf] 2024-12-05T14:21:38.045Z	INFO	service/init.go:323	server started	{"component": "AMF", "category": "Init"}
```

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

File: tests/unit/expected_config/config.conf [hunk 1]
```diff
@@ -64,6 +64,9 @@ configuration:
     enable: true
     expireTime: 6s
     maxRetryTimes: 4
+logger:
+  AMF:
+    debugLevel: info
```
Comment (Gmerold) [inline tests/unit/expected_config/config.conf:69]: In the JIRA ticket I also proposed setting the `ReportCaller` to true when we go to `debug` mode. It made my life much easier when I was debugging the upstream.
Should we include it this feature?

Comment (gruyaume) [reply tests/unit/expected_config/config.conf:69]: Are you sure reportCaller is still a thing? I can see no reference to it in the upstream logger code:

Reference: https://github.com/omec-project/util/blob/main/logger/logger_config.go

Comment (gruyaume) [reply tests/unit/expected_config/config.conf:69]: Indeed, the report caller config option got removed part of the move from logrus to zap. For reference here's the PR upstream: https://github.com/omec-project/util/pull/45

**Review by Gmerold — APPROVED** [review]

File: tests/unit/expected_config/config.conf (hunk 1 above)
Comment (gatici) [reply tests/unit/expected_config/config.conf:69]: Yes,  
https://github.com/omec-project/util/blob/main/logger/logger.go#L47 adds the caller function and line to the log output. So, this is already available.

**Review by gatici — APPROVED** [review]

**Pull request merged**